aten/src/ATen/core/type.cpp

```diff
@@ -812,6 +812,22 @@ bool ClassType::isSubtypeOfExt(const TypePtr rhs, std::ostream* why_not) const {
   return Type::isSubtypeOfExt(rhs, why_not);
 }
 
+bool ClassType::is_hook(const std::string& fn_name) {
+  QualifiedName qname(fn_name);
+  const auto& name = qname.name();
+  for (const auto& hook : forward_pre_hooks) {
+    if (hook.name() == name) {
```

Comment:
Can't we just use the operator== here ? https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/qualified_name.h#L99

Reply:
Actually, probably not the way you want. `fn_name` here is something like `M.hook_fn` - that's how we store it in the call stack - whereas `hook` itself is `hook_fn` - hooks don't really use fully qualified names as there is no need. And the real qualified name for these would be `__torch__.M.hook_fn`. I can use the `QualifiedName::operator==`, but then I would have to prepare the data kinda the same way:

```
  QualifiedName qname(fn_name);
  QualifiedName name(qname.name());
  for (const auto& hook : forward_pre_hooks) {
    if (hook == name) {
```
I can stop using `QualifiedName`s for hooks it that will make it easier to follow.